conda_smithy/configure_feedstock.py

```diff
@@ -482,16 +482,76 @@ def _render_ci_provider(
     for i, (platform, arch, keep_noarch) in enumerate(
         zip(platforms, archs, keep_noarchs)
     ):
-        metas = conda_build.api.render(
-            os.path.join(forge_dir, "recipe"),
+        config = conda_build.config.get_or_merge_config(None,
             exclusive_config_file=forge_config["exclusive_config_file"],
             platform=platform,
             arch=arch,
-            permit_undefined_jinja=True,
-            finalize=False,
-            bypass_env_check=True,
-            channel_urls=forge_config.get("channels", {}).get("sources", []),
         )
+
+        old_combine_specs = conda_build.variants.combine_specs
+        def combine_specs_and_apply_migrations(specs, log_output=True):
+            """CFEP-9 variant migrations
+
+            Apply the list of migrations configurations to the build (in the correct sequence)
+            This will be used to change the variant within the list of MetaData instances,
+            and return the migrated variants.
+
+            This has to happend before the final variant files are computed.
+
+            The method for application is determined by the variant algebra as defined by CFEP-9
+
+            """
+            # Original call
+
+            combined_spec_res = old_combine_specs(specs, log_output=log_output)
+            if isinstance(combined_spec_res, tuple):
+                combined_spec, extend_keys = combined_spec_res
+            else:
+                combined_spec = combined_spec_res
+
+            migrations_root = os.path.join(forge_dir, "migrations", "*.yaml")
+            migrations = glob.glob(migrations_root)
+
+            from .variant_algebra import parse_variant, variant_add
+
+            migration_variants = [
+                (fn, parse_variant(open(fn, "r").read(), config=config)) for fn in migrations
+            ]
+            migration_variants.sort(key=lambda fn_v: (fn_v[1]["migration_ts"], fn_v[0]))
+            if len(migration_variants):
+                print(f"Applying migrations: {','.join(k for k, v in migration_variants)}")
+
+            for migrator_file, migration in migration_variants:
+                if 'migration_ts' in migration:
+                    del migration['migration_ts']
+                if len(migration):
+                    combined_spec = variant_add(combined_spec, migration)
+
+            # Avoid spec based filtering that is done by default.
+            for k in specs.keys():
+                if k == 'internal_defaults':
+                    continue
+                del specs[k]
+
+            # Deal with legacy return value
+            if isinstance(combined_spec_res, tuple):
+                return combined_spec, extend_keys
+            else:
+                return combined_spec
+
+        try:
+            conda_build.variants.combine_specs = combine_specs_and_apply_migrations
```

Comment:
@msarahan This is the method we are patching for making CFEP-9 migrations work.  Would this be a suitable addition / enhancement to conda-build to provide a hook here?